enrollment-server-domain-model/src/main/java/com/wultra/app/enrollmentserver/model/integration/OwnerId.java

```diff
@@ -69,10 +68,9 @@
      * @return Securely hashed user identification.
      * This can be used to hide the original possibly sensitive identity value.
      */
-    @Nullable
     public String getUserIdSecured() {
         if (userId == null) {
-            return null;
+            throw new IllegalStateException("Missing userId value");
```

Comment:
Hmm, I guess that's not true, have you tested this?

The `IProovPresenceCheckProvider` will catch any exception and forward it as `PresenceCheckException` which are handled in our `DefaultExceptionHandler`. I am not sure we should forward the real cause at this case as the client should not be aware of our technical solution.

There is similar pattern for any unexpected exceptions.

I always use these runtime exceptions for such exceptional failures and rely on upper global exception handling.

The only strange thing could be that it will keep failing unless we will have some limitation on repeated attempts.

Reply:
Ah, ok, I have overlooked this handling. You can ignore the comment.